src/ocnicepost.fd/outputvars.F90

```diff
@@ -0,0 +1,146 @@
+module outputvars
+
+  implicit none
+
+  integer, parameter :: maxvars = 40          !< The maximum number of variables written to a file
+
+  type :: vardefs
+     character(len= 10)   :: input_var_name    !< A variable's  input variable name
+     character(len= 10)   :: output_var_name   !< A variable's output variable name
+     character(len=120)   :: long_name         !< A variable's long name
+     character(len= 20)   :: units             !< A variable's unit
+     character(len= 10)   :: var_remapmethod   !< A variable's mapping method
+     integer              :: var_dimen         !< A variable's dimensionality
+     character(len=  4)   :: var_grid          !< A variable's input grid location; all output locations are on cell centers
+     character(len= 10)   :: var_pair          !< A variable's pair
+     character(len=  4)   :: var_pair_grid     !< A pair variable grid
+     real                 :: var_fillvalue     !< A variable's fillvalue
+  end type vardefs
+
+  type(vardefs), public :: outvars(maxvars)    !< Attribute definitions for the variables
+
+contains
+
+  subroutine ocnvars_typedefine
+
+    ! local variables
+    integer :: ii = 0
+
+    !set defaults
+    outvars(:)%input_var_name=''
```

Comment:
This entire subroutine needs to be defined via a namelist or CSV file. Nothing should be hard coded as it limits the capabilities of the software.

Reply:
I'm not sure I understand the comments regarding generality. Variables in MOM6/CICE6 output have specified names, on specified grid locations. MOM6 temperature is never written in the output file on anything other than the center grid locations.

Reply:
I think @HenryWinterbottom-NOAA's point is to make this configurable so that one can add/remove variables without having to rebuild executables.  It separates action (code) from options (configuration).

This can be addressed later as an improvement activity.